Added support for custom commands

leandropineda opened the pull request:
Added support for custom commands and reporting status back to the server.
Note that the major version has been updated since the backend introduced support for custom commands for agent/SDK version >1.2.2.

[Peek 2022-08-19 15-40.webm](https://user-images.githubusercontent.com/8092356/185685928-e66998a9-d164-4baa-bb1f-5f38dc031ed2.webm)

miguelgarcia commented on src/index.js:
```diff
@@ -188,7 +191,24 @@ class RobotSession {
+    const msg = new messages.CustomScriptCommandMessage.deserializeBinary(message);
+    // Hand over to callback for processing, using the proper format
+    this.#dispatchCommand(
+      'customCommand',
```
nit. move to exported constant

miguelgarcia commented on src/index.js:
```diff
@@ -197,20 +217,25 @@ class RobotSession {
+    const msg = new messages.CustomScriptStatusMessage();
+    msg.setFileName(args[0]);
+    msg.setExecutionId(executionId);
+    msg.setExecutionStatus((resultCode === '0' ? 'finished' : 'aborted'));
```
`finished` and `aborted` should be constants

miguelgarcia commented:
Have you tested it ? Would you include a demo video showing that the results appear in the Platform ?

miguelgarcia reviewed (commented):
Demo would be great

leandropineda commented:
@miguelgarcia I addressed all the comments and added a demo video.

I also refactor the example files a little so they can be executed from the same repository folder. WDYT?

miguelgarcia reviewed (approved):
:+1:  LGTM!
